python/tank/templatekey.py

```diff
@@ -292,31 +304,164 @@ def __init__(self,
                                          abstract=abstract,
                                          length=length)
 
-        if not(format_spec is None or isinstance(format_spec, basestring)):
-            msg = "Format_spec for TemplateKey %s is not of type string: %s"
+    def _init_format_spec(self, name, format_spec):
+        """
+        Asserts that the format_spec parameter is a valid value.
+
+        :param name: Name of this template key.
+        :param format_spec: Parameter to be validated.
+
+        :raises TankError: Raised when the parameter is not a string that maching a %d format
+                           option.
+        """
+        # No format spec means no formatting options.
+        if format_spec is None:
+            self._padding_char = None
+            self._minimum_width = None
+            self.format_spec = None
+            return
+
+        if isinstance(format_spec, basestring) is False:
+            msg = "format_spec for IntegerKey %s is not of type string: %s"
             raise TankError(msg % (name, str(format_spec)))
```

Comment:
No need to cast to str if you use %s in the msg
Also, this feels a lot easier to read:

```
raise TankError("format_spec for IntegerKey %s is not of type string: %s" % (name, format_spec))
```

Also, maybe we could just implement a `__repr__` on the `TemplateKey` level and then just do:

```
raise TankError("Format Spec for %s is not a string: %s" % (self, format_spec))
```

That seems like a useful addition to me!